nifi/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java

```diff
@@ -239,87 +128,39 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
+        final ObjectHolder<String> extensionRef = new ObjectHolder<>(null);
         session.read(flowFile, new InputStreamCallback() {
             @Override
             public void process(final InputStream stream) throws IOException {
                 try (final InputStream in = new BufferedInputStream(stream)) {
-                    // read in up to magicHeaderMaxLength bytes
-                    in.mark(magicHeaderMaxLength);
-                    byte[] header = new byte[magicHeaderMaxLength];
-                    for (int i = 0; i < header.length; i++) {
-                        final int next = in.read();
-                        if (next >= 0) {
-                            header[i] = (byte) next;
-                        } else if (i == 0) {
-                            header = new byte[0];
-                        } else {
-                            final byte[] newBuffer = new byte[i - 1];
-                            System.arraycopy(header, 0, newBuffer, 0, i - 1);
-                            header = newBuffer;
-                            break;
-                        }
-                    }
-                    in.reset();
-
-                    for (final MagicHeader magicHeader : magicHeaders) {
-                        if (magicHeader.matches(header)) {
-                            mimeTypeRef.set(magicHeader.getMimeType());
-                            return;
-                        }
-                    }
-
-                    if (!identifyZip) {
-                        for (final MagicHeader magicHeader : zipMagicHeaders) {
-                            if (magicHeader.matches(header)) {
-                                mimeTypeRef.set(magicHeader.getMimeType());
-                                return;
-                            }
-                        }
-                    }
-
-                    if (!identifyTar) {
-                        for (final MagicHeader magicHeader : tarMagicHeaders) {
-                            if (magicHeader.matches(header)) {
-                                mimeTypeRef.set(magicHeader.getMimeType());
-                                return;
-                            }
-                        }
+                    TikaInputStream tikaStream = TikaInputStream.get(in);
+                    Metadata metadata = new Metadata();
+                    // Get mime type
+                    MediaType mediatype = detector.detect(tikaStream, metadata);
+                    mimeTypeRef.set(mediatype.toString());
+                    // Get common file extension
+                    try {
+                        MimeType mimetype;
+                        mimetype = config.getMimeRepository().forName(mediatype.toString());
+                        extensionRef.set(mimetype.getExtension());
+                    } catch (MimeTypeException ex) {
+                        logger.warn("MIME type detection failed: {}", new Object[]{ex.toString()});
```

Comment:
I would use "new Object[] {ex}" rather than "new Object[] {ex.toString()}"... this way, if the user configures logback to use debug-level logging, the logger will automatically get the stack trace.